Source-Code/FlareDraw.cs

```diff
@@ -192,9 +192,22 @@ private void GenerateBodyFlares()
             //ListChildren(sm.systemPrefab.rootBody, 0);
             //--- HACK--
 
+            //If Kerbin is parented to the Sun, set its SMA- otherwise iterate
+            //through celestial bodies to locate which is parented to the Sun
+            //and has Kerbin as a child. Set the highest parents SMA to kerbinSMA
             if(BodyFlare.kerbinSMA <= 0.0)
             {
-                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
+                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                else
+                {                    
+                    foreach(CelestialBody current in FlightGlobals.Bodies)
```

Comment:
No big reason, that would just assume that Kerbin's parent would be
directly parented to the Sun. I'm not actually sure if its possible, or
somebody even would, parent something like Sun>Something>Something2>Kerbin.
The children check is whether or not it has it as a child regardless of any
sub parenting levels so you get the SMA of the primary body under the Sun.
I actually wanted to use a for() statement that broke out when it was
located but wasn't sure how to find how many bodies were in the index for
the range.

I guess I just thought it was easier to setup checking for the primary
parent body instead of pickwalking from Kerbin up to the top level.

On Thu, Aug 20, 2015 at 8:21 AM, MOARdV <notifications@github.com> wrote:

> In Source-Code/FlareDraw.cs
> <https://github.com/MOARdV/DistantObject/pull/20#discussion_r37522499>:
>
> >              if(BodyFlare.kerbinSMA <= 0.0)
> >              {
> > -                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
> > +                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
> > +                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
> > +                else
> > +                {
> > +                    foreach(CelestialBody current in FlightGlobals.Bodies)
>
> This looks fine. Is there a reason you couldn't check the .referenceBody
> field directly (that is, in the else block get Bodies[1].referenceBody, and
> verify its parent is the sun, and use its .orbit field)? That'd be more
> efficient, assuming there's not something that prevents it from working
> that I'm overlooking.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MOARdV/DistantObject/pull/20/files#r37522499>.
>